pvlib/pvsystem.py

```diff
@@ -2537,6 +2553,73 @@ def pvwatts_losses(soiling=2, shading=3, snow=0, mismatch=2, wiring=2,
     return losses
 
 
+def irradiance_loss_pvsyst(effective_irradiance, irradiance_shading_loss,
+                           irradiance_snow_loss, irradiance_soiling_loss):
+    r"""
+    Calculates plane of array irradiance losses using a model based on the
+    PVSyst methodology [1]_ of linearly reducing the effective irradiance on a
+    module.  This doesn't account for the electrical losses due to irregular
+    irradiance on different solar cell strings but rather reduces the
+    irradiance input to the single diode equation.
+
+    The losses are compounded using the following equation:
+
+    .. math::
+
+        L_{total}(\%) = 100 [ 1 - \Pi_i ( 1 - \frac{L_i}{100} ) ]
+
+    Note the parameters must each be a series with a DatetimeIndex.  The
+    index of the returned series will be the same as the effective_irradiance
+    index.  All other parameters will be resampled to match this index with
+    a "fill forward" method for filling holes.
+
+    Parameters
+    ----------
+    effective_irradiance : Series
+            The irradiance (W/m2) that is to be reduced.  Plane-of-array
+            irradiance that already accounts for Angle of Incidence and
+            Spectral losses.
+
+    irradiance_shading_loss : Series
+            The fraction of the solar panel that is shaded.
+
+    irradiance_snow_loss : Series
+            The fraction of the solar panel that is covered with snow.
+
+    irradiance_soiling_loss : Series
+            The fraction of the irradiance prevented from reaching the solar
+            cells due to soiling of the solar panel.
```

Comment:
These inputs are in %, from the equation. If that's correct, change "fraction" to "percent", or add [%] at the end of the description.

Reply:
I realize now that the math in my docstring doesn't match my code.  With the code, I intended to have the parameters input as fractions (input .02 to mean a 2% reduction in irradiance) and the output as a loss of irradiance in W/m2.  The docstring equation implies an input of percentages (input 2 to mean a 2% reduction in irradiance) and an output as a percentage of irradiance (which means the effective_irradiance parameter isn't needed anymore).

I'm assuming the preference is just that I pick one or the other and stay consistent throughout, but I thought this would be a good thing to ask for an opinion on, as it will guide the way we approach all of the other loss models going forward.  For reference, the current pvlib spectral/aoi loss models output loss coefficients and the pvwatts_loss_model uses the methodology spelled out in the docstring (returns a percentage).  @cwhanse any preference?

Reply:
I'm sure that the processes models for e.g. reflection loss (in `pvlib.iam`) return fractions, as does the snow model in `pvlib.snow` and the soiling model. 

Loss parameters in `PVSystem.pvwatts_losses` are in percent and the output is an overall loss factor in %. Unless there's a good reason to change to fractions (and modify the existing function) I'm OK staying with % as the unit for loss model parameters and for it's output.

I suspect that the PVsyst loss chain has a lot in common with the PVWatts loss chain. That's OK, one mission of pvlib is to clarify what different models do.  At some point we probably should have a utility function to convolve loss factors that is used by the two loss chains, perhaps private. Doesn't have to be in this PR.